app/src/full/java/io/homeassistant/companion/android/sensors/HealthConnectSensorManager.kt

```diff
@@ -0,0 +1,168 @@
+package io.homeassistant.companion.android.sensors
+
+import android.content.Context
+import android.os.Build
+import android.os.Build.VERSION.SDK_INT
+import androidx.health.connect.client.HealthConnectClient
+import androidx.health.connect.client.permission.HealthPermission
+import androidx.health.connect.client.records.ActiveCaloriesBurnedRecord
+import androidx.health.connect.client.records.TotalCaloriesBurnedRecord
+import androidx.health.connect.client.records.WeightRecord
+import androidx.health.connect.client.request.AggregateRequest
+import androidx.health.connect.client.request.ReadRecordsRequest
+import androidx.health.connect.client.time.TimeRangeFilter
+import io.homeassistant.companion.android.common.R as commonR
+import io.homeassistant.companion.android.common.sensors.SensorManager
+import java.time.Instant
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.ZoneOffset
+import java.time.temporal.ChronoUnit
+import kotlinx.coroutines.runBlocking
+
+class HealthConnectSensorManager : SensorManager {
+    companion object {
+        val activeCaloriesBurned = SensorManager.BasicSensor(
+            id = "health_connect_active_calories_burned",
+            type = "sensor",
+            commonR.string.basic_sensor_name_active_calories_burned,
+            commonR.string.sensor_description_active_calories_burned,
+            "mdi:fire",
+            unitOfMeasurement = "kcal"
+        )
+
+        val totalCaloriesBurned = SensorManager.BasicSensor(
+            id = "health_connect_total_calories_burned",
+            type = "sensor",
+            commonR.string.basic_sensor_name_total_calories_burned,
+            commonR.string.sensor_description_total_calories_burned,
+            "mdi:fire",
+            unitOfMeasurement = "kcal"
+        )
+
+        val weight = SensorManager.BasicSensor(
+            id = "health_connect_weight",
+            type = "sensor",
+            commonR.string.basic_sensor_name_weight,
+            commonR.string.sensor_description_weight,
+            "mdi:weight",
+            unitOfMeasurement = "g"
```

Comment:
we should use the `weight` device class here :)

https://www.home-assistant.io/integrations/sensor/#device-class

should allow users to select the proper unit they prefer because grams.... 😂

Reply:
Added 👍